src/main/scala/com/fulcrumgenomics/util/AyncStreamSink.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.fulcrumgenomics.util
+
+import java.io.{Closeable, InputStream}
+import java.util.concurrent.atomic.AtomicInteger
+import dagr.commons.CommonsDef._
+
+import scala.io.Source
+
+object AsyncStreamSink {
+  private val n: AtomicInteger = new AtomicInteger(1)
+  private def nextName: String = "AsyncStreamSinkThread-" + n.getAndIncrement
+}
+
+/**
+  * Class that is capable of reading output from a sub-process (or arbitrary other InputStream)
+  * and processing it through any method that accepts a String, while managing the extra thread
+  * necessary to ensure nothing blocks.
+  */
+class AsyncStreamSink(in: InputStream, private val sink: String => Unit) extends Closeable {
+  private val source = Source.fromInputStream(in).withClose(() => in.close())
+  private val thread = new Thread( new Runnable { def run() : Unit = { source.getLines().foreach(sink) } } )
```

Comment:
Yes, it returns an iterator that will keep returning lines when invoked - block if necessary - and then self-close when it hits EOF.